src/polygon.c

```diff
@@ -850,6 +852,54 @@ spherepoly_in(PG_FUNCTION_ARGS)
 	PG_RETURN_POINTER(poly);
 }
 
+Datum spherepoly_deg(PG_FUNCTION_ARGS)
+{
+	int num_elements, i, point_index, np;
+	ArrayType *float_vector;
+	float8 *array_data;
+	SPOLY *spoly;
+	SPoint *points;
+
+	float_vector = PG_GETARG_ARRAYTYPE_P(0);
+	num_elements = ArrayGetNItems(
+		ARR_NDIM(float_vector), ARR_DIMS(float_vector)
+	);
+	np = num_elements / 2;
+
+	if (num_elements <= 0)
+	{
+		PG_RETURN_NULL();
+	}
+
+	if (!(num_elements >= 6) || !(num_elements % 2 == 0))
+	{
+		elog(ERROR, "Invalid arguments count. It must be even and >= 6");
```

Comment:
The other error messages in this file all have the name of the function as a prefix to the error. So I think something like `spherepoly_deg: invalid number of arguments (must be even and >= 6)` would fit the style better.